system-test/bigquery.ts

```diff
@@ -28,7 +28,7 @@ const bigquery = new BigQuery();
 const storage = new Storage();
 
 describe('BigQuery', () => {
-  const GCLOUD_TESTS_PREFIX = 'nodejs_bq_test';
+  const GCLOUD_TESTS_PREFIX = `bq${Date.now()}`;
```

Comment:
Can we keep some `nodejs_` reference in there. This helped us not collide with other projects, as well as just being friendlier when we need to trace where a straggler came from.

Reply:
Fair enough, we'll probably have to change the way we generate the resource names via `generateName` though because we're pretty much at the limit just by adding the timestamp.

Reply:
In other libs we're doing a lot of `gcloud-test-` or something to that effect.

Reply:
We were getting in trouble with the generic prefix when our before/after hooks filtered previously-created resources. It would actually delete the other projects'! I believe we added in some more logic about how old a resource must be before we consider it dead, but I'm not sure which repos we have done that in. At a minimum, a prefix unique to the language of the project eliminates most chances for collisions.

Reply:
Oh bother - I meant something like `gcloud-test-${uuid.v4().split('-')[0]}`

Reply:
The collision potential is still there, because our test hooks filter on the first part of the prefix (the preprefixfix, in technical terms), like `gcloud-test-`, so that we can get a list of all resources created by the project without knowing the full name.

Reply:
So I've removed including the resource type in the name and it would appear we're staying under the limit.

Reply:
That's good except for test failure, cause it can be super confusing figuring out what "Cannot delete ___random_name__" means :(

Reply:
Where would that message come from though? In BQ we only delete Buckets and Datasets, I'm not particularly familiar with what error messages would be for either, but doing a quick test Storage doesn't seem to include the bucket name for not found or permission errors.

Reply:
Hmm, I'm not sure specifically the errors for these libs, but I recall it being very important with Compute, because many different resource types are deleted within the same after hook, and it says "blah cannot be deleted". Can we just clip more off of the uuid? We can play with the prefix, too, e.g. `bqnode_{type}_{uuid}`

Reply:
Ok, made a new commit. The hard limit for Buckets is 63 so I just added a `.substr(0, 63)` to the end of the generated name.

Reply:
Another easy idea - for the randomized part, use `Math.random().toString(36).slice(2)`  👅